src/RemoteAPI/class-analytics-post-detail-api.php

```diff
@@ -23,10 +24,19 @@ class Analytics_Post_Detail_API extends Base_Endpoint_Remote {
 	protected const QUERY_FILTER = 'wp_parsely_analytics_post_detail_endpoint_args';
 
 	/**
-	 * Indicates whether the endpoint is public or protected behind permissions.
+	 * Returns whether the endpoint is available for access by the current
+	 * user.
 	 *
-	 * @since 3.7.0
-	 * @var bool
+	 * @since 3.14.0
+	 *
+	 * @return bool
 	 */
-	protected $is_public_endpoint = false;
+	public function is_available_to_current_user(): bool {
```

Comment:
Since most of the times the wanted behavior is to check for the `Base_Endpoint::DEFAULT_ACCESS_CAPABILITY` capability, would it make sense to implement this function in the [Base_Endpoint class](https://github.com/Parsely/wp-parsely/blob/2298ad122f1d6ea34dbbfdffd732bda95dfccd66/src/Endpoints/class-base-endpoint.php#L63) and have it be the default behavior?

Reply:
Thanks for chiming in about this. I didn't set a default behavior on purpose, in order to force the implementation in child classes. The idea was to be more explicit about it since we're setting access permissions. I agree it does duplicate some lines of code though.